.github/workflows/raspberry-pi.yaml

```diff
@@ -0,0 +1,93 @@
+name: Build and test on Virtualized Raspberry Pi OS
+
+on:
+  push:
+    branches: [ master, develop ]
+  pull_request:
+    branches: [ master, develop ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+        include:
+          # Debian 11
+          - os: bullseye
+            image: ghcr.io/dtcooper/raspberrypi-os:python3.12-bullseye
+          # Debian 12
+          - os: bookworm
+            image: ghcr.io/dtcooper/raspberrypi-os:python3.12-bookworm
+      fail-fast: false
+
+    name: Build on ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-session-name: ${{ secrets.AWS_ROLE_SESSION_NAME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-duration-seconds: 10800
+
+      - name: Build and Test
+        env:
+          AWS_KVS_LOG_LEVEL: 2
+        run: |
+          docker run --rm -v ${{ github.workspace }}:/workspace -w /workspace \
+            -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY \
+            -e AWS_SESSION_TOKEN -e AWS_REGION -e AWS_KVS_LOG_LEVEL \
+            --platform linux/arm64 \
+            ${{ matrix.image }} \
+            /bin/bash -c '
+              set -ex
+              apt-get update
+              apt-get install -y automake build-essential cmake git \
+                gstreamer1.0-plugins-base-apps gstreamer1.0-plugins-bad \
+                gstreamer1.0-plugins-good gstreamer1.0-plugins-ugly \
+                gstreamer1.0-tools gstreamer1.0-omx-generic \
+                libcurl4-openssl-dev libgstreamer1.0-dev \
+                libgstreamer-plugins-base1.0-dev liblog4cplus-dev \
+                libssl-dev pkg-config
+
+              mkdir -p build
+              cd build
+
+              cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_DEPENDENCIES=OFF -DALIGNED_MEMORY_MODEL=ON
+              make -j$(nproc)
+
+              export GST_PLUGIN_PATH=$(pwd)
```

Comment:
Not immediately required for this CR but is something to consider for the future: It would be worth checking if we remove the toolchain after the build has completed successfully.